makelib/golang.mk

```diff
@@ -97,7 +97,7 @@ endif
 
 # We use a consistent version of golangci-lint to ensure everyone gets the same
 # linters.
-GOLANGCILINT_VERSION := 1.15.0
+GOLANGCILINT_VERSION := 1.22.2
```

Comment:
there were some massive perf and resource usage issues for golangci-lint like 1.17.x or something, are all of those solved and this 1.22.2 looks great during CI build runs?

Reply:
1.15.0 was really broken with go 1.13. I can only say that my build passed with this combo, but can't speak to if that works in all repos. I assume actually we'll catch some bugs with this better linter on this version, so as people update the build sub-module I expect a small amount of bugfixes. @hasheddan did mention that he's been using 1.18 I think, but I've updated the PR to make this GOLANGCILINT_VERSION configurable in case someone has a preferred version.

Reply:
Yep anything 1.18+ has been working fine for me.